### ISSUE TITLE
Request validation fails when POSTing an array with only one element

### ISSUE THREAD

**Issue by workshur** [apigee-127/sway#140]
```yml
/tags:
    x-swagger-router-controller: tags
    post:
      operationId: addTags
      parameters:
        - in: body
          name: tags
          required: true
          schema:
            type: array
            items: 
              type: number
              format: int32
      responses:
        "200":
          description: Success
        default:
          $ref: '#/responses/ErrorResponse'
```
This simple definition works good for empty arrays and arrays with multiple items. 
```
curl -H "Content-Type: application/json" -X POST -d '[]' http://localhost:8080/tags
curl -H "Content-Type: application/json" -X POST -d '[1,2]' http://localhost:8080/tags
```

But it fails for array with only one element.
```
curl -H "Content-Type: application/json" -X POST -d '[1]' http://localhost:8080/tags
```
```js
{ 
  code: 'INVALID_TYPE', 
  params: ['array', 'integer'], 
  message: 'Expected type array but found type integer', 
  path: [] 
}
```

As I understand it, the reason lies [here](https://github.com/apigee-127/sway/blob/e53e529dc4a01b11a40f9e968fc15be76a292c73/lib/helpers.js#L213):
```js
  // Attempt to parse the string as JSON if the type is array or object
  if (['array', 'object'].indexOf(type) > -1) {
    try {
      value = JSON.parse(value);
    } catch (err) {
      // Nothing to do here, just fall through
    }
  }
```
`JSON.parse([1]);` returns integer 1

I also see that we have a [handler](https://github.com/apigee-127/sway/blob/e53e529dc4a01b11a40f9e968fc15be76a292c73/lib/helpers.js#L232) for such situations. It will work If I add `collectionFormat: multi`. But as I see in http://swagger.io/specification/, we can use `collectionFormat` only "If `in` is any value other than `body`".

**Comment by whitlockjc** [comment]
This has been fixed here: https://github.com/apigee-127/sway/pull/114  I'll be putting together a release this week that should fix this.

**Comment by workshur** [comment]
@whitlockjc Wow. Sounds good. Many thanks

**Comment by noahdietz** [comment]
Need #114 to be fixed (just reseting some conflicting changes) then we can merge that to fix this...

**Comment by whitlockjc** [comment]
This was fixed using 94ba34fe8008e9febdff2ff1ee48027ac95a881f.
